File: 1741-sort-array-by-increasing-frequency/1741-sort-array-by-increasing-frequency.java
class Solution {
        public int[] frequencySort(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        List<Map.Entry<Integer, Integer>> list = new ArrayList(map.entrySet());
        Collections.sort(list, (a,b) -> a.getValue() == b.getValue() ? b.getKey() - a.getKey() : a.getValue() - b.getValue());
        int index = 0;
        int[] res = new int[nums.length];
        for (Map.Entry<Integer, Integer> entry : list) {
            
            int count = entry.getValue();
            int key = entry.getKey();
            
            for (int i=0; i<count; i++) {
                res[index++] = key;
            }
        }
        return res;
    }
}

File: 1741-sort-array-by-increasing-frequency/1741-sort-array-by-increasing-frequency.py
class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        c=Counter(nums)
        c=sorted(c.items(),key=lambda x:(x[1],-x[0]))
        ans=[]
        for x in c :
            ans +=[x[0] for i in range(x[1])]
        return ans